### DISCUSSION TITLE
RNS Presentation

### DISCUSSION THREAD

**Discussion by r8io** [markqvist/Reticulum#233] [Show and tell]
Hello Mark, hello Reticulum enthusiasts,

I am in the process of learning about the Reticulum Network Stack. What better idea is there, than to produce presentations about a subject you are not familiar with :-)

I made a little introductory presentation about Reticulum which can be viewed here [RNS Presentations](https://r8io.github.io/rns-presentations)

Please feel free to like, dislike or discuss.

**Comment by markqvist** [comment]
Wow! This is really good @r8io. You really managed to collect all the relevant info and condense it into a nice and comprehensible introductory overview. I don't even know how you managed to get it all collected, so much of it is still scattered all over the place ;)

I have a few clarifications I'd like to add, if you don't mind? Would you rather have them as pull requests or  just in loose form here?

**Reply by r8io** [reply]
Mark, thank you for your kind words! I am glad the presentation might be helpful.

For the clarifications any form that you prefer is fine with me. Please let me know what should be changed or improved and I will get right to it.

**Comment by markqvist** [comment]
In slide 10:
- There's a typo: `Nomade Network instance` -> `Nomad Network instance`
- Maybe also mention the Sideband app here, and that both Nomad Network and Sideband are LXMF clients. Maybe this is redundant, since you also mention it later. Your call what is best here, just wanted to mention it.
- Maybe also mention that anyone can make programs and apps that interoperate with each other over LXMF, it is a very open and easy to use system.

In slide 16:
- There's a typo: `Goodby` -> `Goodbye`

In slide 23:
- The RNode firmware is not written in Python, but in C.

In slide 24:
- Not a correction, but just want to say good choice for mentioning GrapheneOS as reference client. This is the first and primary target I use for testing everything made for Android-based platforms with.

Thanks again for putting this together!

**Reply by r8io** [reply]
Thank you for your feedback,

* Fixed the typos
* Changed the RNode implementation to C.
* Tried to improve slide 10 as suggested, but running out of time for today. I'll get back to it as soon as possible. Please feel free to comment :-)
* I will add another slide that points out that "anyone can make programs and apps that interoperate with each other over LXMF".

I am glad you liked the mentioning of GrapheneOS, as I tried to seek it in under the radar ;-)

Cheers and thank you for making this all possible!

**Comment by r8io** [comment]
Short version:
I finished incorporating your feedback into the [slides](https://r8io.github.io/rns-presentations/source/001-introduction.html). For details please see the [Release Notes](https://github.com/r8io/rns-presentations/releases/tag/0.1.0). If time permits, please let my know what you think.

Long version:
I gave my little introductory presentation on Wednesday to a couple of guys with various backgrounds, from novice to professional. We started out with scanning a QR paper message. After that I introduced a basic setup with a phone running Sideband + RNode on one side and a laptop running Sideband and Nomadnet + Rnode on the other side. We exchanged some messages and I showed Nomadnet's capability to display a page.

The file transfer from a node running on a VM to the node running on my host system didn't go so well, probably because the network connection wasn't working as I assumed it would. I guess I should have tested my setup more thoroughly. But no worries, I was _easily_ able to convince my little audience that the problem was on my side and that it had nothing to do with Reticulum, so we moved on to the slides :-)

A point of discussion was the fact that Reticulum only needs the physical layer, which provides the raw bit stream, to be functional. It took a while to clarify that MAC addresses, which reside on the data link layer, are not needed by Reticulum. I tried to improve slide 18 `Goodbye Internet Stack` to show this information more clearly.

Overall I think the presentation was quite well received. For some guys the information was a little overwhelming, others seemed to comprehend that Reticulum is indeed a glorious endeavor. We will see what happens and if I succeeded to get some more people to try out Reticulum.

**Reply by markqvist** [reply]
Really appreciate you taking the time to relay this story and the experience of it. This is the kind of communication work that is needed to get a broader understanding of Reticulum out there :)
